### PR TITLE
feat: introduce StatusUserAccount type

### PR DESCRIPTION
**Pull request by MatousJobanek**
to make the comparison/navigation easier

### CODE REVIEW

File: pkg/apis/toolchain/v1alpha1/useraccount_types.go [hunk 1]
```diff
@@ -35,7 +36,7 @@ type UserAccountStatus struct {
 	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
 
 	// Observed status. For example: provisioning|provisioned
-	Status string `json:"status,omitempty"`
+	Status StatusUserAccount `json:"status,omitempty"`
```
Comment (nurali-techie) [inline pkg/apis/toolchain/v1alpha1/useraccount_types.go]: hope `make generate` target run (to generate stuff in member/host operator) and things are validated to make sure member/host operator are fine with the changes ??

Comment (xcoulon) [reply pkg/apis/toolchain/v1alpha1/useraccount_types.go]: @nurali-techie `make generate` dispatches the CRD files in the `member-operator` and `host-operator` repositories, but it won't run the `make build` or `make test` in those repo.

Comment (nurali-techie) [reply pkg/apis/toolchain/v1alpha1/useraccount_types.go]: I am talking about manual_validation after running `make generate`.

Comment (MatousJobanek) [reply pkg/apis/toolchain/v1alpha1/useraccount_types.go]: I probably don't understand your question. `make generate` just generates and dispatches the CRDs - that doesn't have any impact on the code... Anyway, this change that is introduced in this PR shouldn't change the CRD anyhow - the value is still string (verified locally ;-)).

**Pull request merged**